### DISCUSSION TITLE
Document creating and using a sidecar docker service

### DISCUSSION THREAD

**Discussion by nipuna-perera** [dagger/dagger#12089] [Ideas]
### What is the issue?

It will be useful to have an example cookbook for adding a docker `dind` sidecar. It is slightly different from the other service examples in the cookbooks as you have to add the `DOCKER_HOST` to the consumer for it to work. The only example is in the dagger code here but it is set to invalidate cache every run - https://github.com/dagger/dagger/blob/aacce090406f99fed23c969fcef769d532f2b587/core/integration/container_test.go#L3260

More details in this discussion - https://discord.com/channels/707636530424053791/1138591097770233986

I think this will be a common use case for certain unit testing scenarios like using localstack to run AWS resource tests for Java.

**Comment by gerhard** [comment]
Yes, I think so too! cc @vikram-dagger
